contrib/pkg/provision_aws/command.go

```diff
@@ -0,0 +1,233 @@
+
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	capiclient "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+
+	clustopclient "github.com/openshift/cluster-operator/pkg/client/clientset_generated/clientset"
+)
+
+type ProvisionClusterOptions struct {
+	Namespace           string
+	Name                string
+	ImageFormat         string
+	RPMRepositoryURL    string
+	RPMRepositorySource string
+	SSHSecretName       string
+	AccountSecretName   string
+	KeyPairName         string
+	AnsibleImage        string
+	Region              string
+	KubeconfigFile      string
+	Version             string
+	VPCName             string
+	AMIVPCName          string
+	LogLevel            string
+	EtcdImage           string
+
+	ClusterName string
+	ImageID     string
+	Logger      log.FieldLogger
+}
+
+func NewProvisionClusterCommand() *cobra.Command {
+	userName := "user"
+	u, err := user.Current()
+	if err == nil {
+		userName = u.Username
+	}
+
+	opt := &ProvisionClusterOptions{}
+	cmd := &cobra.Command{
+		Use:   "provision-aws NAME",
+		Short: "Provision a new cluster with the given RPM repository and Image Format on AWS",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(args); err != nil {
+				log.WithError(err).Error("Cannot complete command")
+				return
+			}
+			if err := opt.Validate(); err != nil {
+				log.WithError(err).Error("Invalid command options")
+				return
+			}
+			if err := opt.Run(); err != nil {
+				log.WithError(err).Error("Runtime error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.Namespace, "namespace", "", "", "namespace to use. Defaults to current namespace.")
+	flags.StringVar(&opt.AccountSecretName, "account-secret", fmt.Sprintf("%s-aws-creds", userName), "secret to use for AWS account")
+	flags.StringVar(&opt.SSHSecretName, "ssh-secret", "libra-ssh", "secret with SSH key")
+	flags.StringVar(&opt.AnsibleImage, "ansible-image", "registry.svc.ci.openshift.org/openshift-cluster-operator/cluster-operator-ansible:latest", "ansible image to use for job")
+	flags.StringVar(&opt.RPMRepositoryURL, "rpm-repository", "", "RPM repository for Origin")
+	flags.StringVar(&opt.RPMRepositorySource, "rpm-repository-source", "https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/releases/openshift/origin/master/origin.repo", "pointer to RPM repository URL")
+	flags.StringVar(&opt.VPCName, "vpc", "", "VPC to use for the cluster. If left blank, a new VPC will be created")
+	flags.StringVar(&opt.AMIVPCName, "ami-vpc", "default", "VPC to use to build the AMI.")
+	flags.StringVar(&opt.LogLevel, "loglevel", "info", "log level, one of: debug, info, warn, error, fatal, panic")
+	// TODO: It is unclear whether this is needed if we're already providing an RPM repo and ImageFormat
+	flags.StringVar(&opt.Version, "origin-version", "3.11", "Version of origin to provision")
```

Comment:
Wondering if this would be better off not origin specific, maybe just openshift-version? (in case this ever got used for ocp, or somebody wanted it to be okd :)